### PR TITLE
Fixes to ConfusionMatrix

### PR DESCRIPTION
**Pull request by kirilllzaitsev**
# Description

- Fix ConfusionMatrix.plot() when classes aren't a list. Addresses https://github.com/roboflow/supervision/issues/212
- Fix input validation to account for empty targets/predictions. Addresses https://github.com/roboflow/supervision/issues/215

## Type of change

Please delete options that are not relevant.

-   [+] Bug fix (non-breaking change which fixes an issue)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Reproduced the issue.

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

### CODE REVIEW

File: supervision/metrics/detection.py [hunk 1]
```diff
@@ -216,20 +216,28 @@ def _validate_input_tensors(
...
-                )
-            if predictions[0].shape[1] != 6:
-                raise ValueError(
-                    f"Predictions must have shape (N, 6). Got {predictions[0].shape} instead."
-                )
-            if targets[0].shape[1] != 5:
-                raise ValueError(
-                    f"Targets must have shape (N, 5). Got {targets[0].shape} instead."
-                )
+            first_non_empty_prediction = next(
```
Comment (SkalskiP) [inline supervision/metrics/detection.py:219]: @kirilllzaitsev, why do we need to look for the first non-empty?

Comment (hardikdava) [reply supervision/metrics/detection.py:219]: Nope,  I found the real issue.

**Review by hardikdava — CHANGES_REQUESTED** [review]
here, if `convert_to_tensor` function is agnostic whether the `detections` or `target`. If it is `target` then `confidences` should not be considered as it will always be `1` and does not influence in result. But `_validation` function is checking fo r shape for `(N,5)` and after adding `confidence` makes shape to `(N,6)`.

**Comment by SkalskiP** [comment]
@hardikdava, your explanation looks reasonable. I'll be working on an alternative fix.

**Comment by SkalskiP** [comment]
@kirilllzaitsev and @hardikdava [here](https://github.com/roboflow/supervision/pull/217) is alternative fix for [#215](https://github.com/roboflow/supervision/issues/215) take a look

File: supervision/metrics/detection.py (hunk 1 above)
Comment (kirilllzaitsev) [reply supervision/metrics/detection.py:219]: @SkalskiP, prediction/target can be either empty or non-empty. Checking the first non-empty instance and assuming that all subsequent non-empty instances will have the same shape, I can raise ValueError if the shape is wrong.

**Comment by SkalskiP** [comment]
@kirilllzaitsev and @hardikdava, I'm doing small housekeeping. I'll close this PR. https://github.com/roboflow/supervision/issues/215 is already fixed. I will submit https://github.com/roboflow/supervision/issues/212 fix later today.

**Pull request closed without merging**